tests/PactNet.Native.Tests/NativePactMessageBuilderTests.cs

```diff
@@ -123,7 +124,7 @@ public void VerifyAsync_Should_Write_Pact_File()
             SetServerReifyMessage(JsonConvert.SerializeObject(content.ToNativeMessage()));
 
             //Act
-            this.messagePact.VerifyAsync<MessageModel>(async _ => { });
+            this.messagePact.VerifyAsync<MessageModel>(async _ => await Task.Run(() => true));
```

Comment:
These would read better as non-async lambdas returning Task.FromResult(true)

I think the test itself should also be async Task instead of void, and then this VerifyAsync would itself be awaited. I'm surprised there aren't more compiler warnings here? Usually if something returns Task but you don't await it then you'll get a warning. In real usage we'd expect callers to be using await here for sure.

Reply:
Yeah, looks like I induced a deadlock by not awaiting anything in the handler. 

"I'm surprised there aren't more compiler warnings here?"

Actually no because the Action or Func can either run sync or async so the compiler is fine with that, but deadlocks at runtime... I had a pair or headaches because of that! 

I will test the verify async in the integration tests (the sample projet) the same way we do for the http interaction model. That will be closer to a real usage.

Reply:
Seems to work fine now